pkg/resources/resource_app_password.go

```diff
@@ -102,98 +178,115 @@ func appPasswordRead(ctx context.Context, d *schema.ResourceData, meta interface
 	}
 
 	client := providerMeta.Frontegg
-	resourceID := d.Id()
+	id := d.Id()
 
-	passwords, err := frontegg.ListAppPasswords(ctx, client)
-	if err != nil {
-		return diag.FromErr(err)
-	}
+	type_ := d.Get("type").(string)
 
-	foundPassword := findAppPasswordById(passwords, resourceID)
-	if foundPassword == nil {
-		d.SetId("")
-		return nil
-	}
+	if type_ == "personal" {
+		tokens, err := frontegg.ListUserApiTokens(ctx, client)
+		if err != nil {
+			return diag.FromErr(err)
+		}
 
-	appPassword := clients.ConstructAppPassword(foundPassword.ClientID, foundPassword.Secret)
+		token := findUserApiTokenById(tokens, id)
+		if token == nil {
+			d.SetId("")
+			return nil
+		}
 
-	// Update the Terraform state with the retrieved values.
-	d.Set("name", foundPassword.Description)
-	d.Set("created_at", foundPassword.CreatedAt.Format(time.RFC3339))
-	d.Set("secret", foundPassword.Secret)
-	d.Set("password", appPassword)
-	// TODO: Get the owner from the API as it's not returned in the response.
-	// d.Set("owner", foundPassword.Owner)
+		// Update the Terraform state with the retrieved values.
+		d.Set("name", token.Description)
+		d.Set("created_at", token.CreatedAt.Format(time.RFC3339))
 
-	return nil
-}
+		// We don't update secret and password because those fields can only be
+		// determined at creation time.
+	} else {
+		// TODO: only fetch the list of SSO roles once per TF provider
+		// invocation.
```

Comment:
Good idea! Just created a tracking issue for this: https://github.com/MaterializeInc/terraform-provider-materialize/issues/579

Reply:
Nice, thanks!